### ISSUE TITLE
Pasting Replaces the SPAN tags in V6.0+ 👍

### ISSUE THREAD

**Issue by TopCoder02** [tinymce/tinymce#10008]
## 📝 Provide detailed reproduction steps (if any)

`<div style="font-size: 0;"><span style="font-size: 16px;"> The span tag will be wiped out in v6.0 if I select all the text and past over it making it invisible</span></div>`

If you highlight all the text in the editor and then paste over it, in v6+ the `<span>` tags with the formatting will be replaced. In version 5.* this doesn't happen. When the text is replaced it is invisible, because of the `font-size:0` on the `<div>` tag.  This only happens with `<span>` tags. 

Please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tiny.cloud](https://fiddle.tiny.cloud) or similar.

1. [Version 6+](https://fiddle.tiny.cloud/v9jF6UGtTP/3)
2. [Version 5.x](https://fiddle.tiny.cloud/v9jF6UGtTP/4) 

### ✔️ Expected result
*What is the expected result of the above steps?*

I expect it to work like version 5.*

### ❌ Actual result
*What is the actual result of the above steps?*
The text is hidden because the `<span>` tags are removed

I'm looking for a workaround or perhaps maybe there is a setting I can add to the .init

**Comment by TheSpyder** [comment]
Which browser are you using? I can't reproduce it in Chrome on macOS.

Note that open source support for TinyMCE 6 [has ended](https://github.com/tinymce/tinymce/discussions/9609), so if this is already fixed in TinyMCE 7 the issue will be closed.

**Comment by TopCoder02** [comment]
@TheSpyder 

I created a video, it's an AI voice over for English.  It should explain the issue.  Yes It happens in Version 7 and I use Chrome on Windows.  But I don't think that matters. This started happening in v6.0, the previous versions didn't have the condition.  It only seems to happen in <span> tags. But I didn't test all HTML tags.. 

https://github.com/user-attachments/assets/90cdbe92-60b2-42bf-a298-a4a8453cbcd2

I hope this conveys, what I wasn't able to convey in writing.

**Comment by TheSpyder** [comment]
Ah! Thank you for that. I thought you meant you had copied from within the editor, not pasted from an external source. That information was a big help.

And it gave me a clue - this _is_ reproducible in TinyMCE 5, with the `paste` plugin. We combined the paste plugin into the core editor with TinyMCE 6 and whatever the problem is came along with it. Here's an updated fiddle with TinyMCE 5 that reproduces the issue:
https://fiddle.tiny.cloud/v9jF6UGtTP/7

This configuration can reproduce the issue all the way back to old versions of TinyMCE 4, I'm afraid. That might make it difficult to track down.

**Comment by TheSpyder** [comment]
The editor without the `paste` plugin worked because it had no paste processing at all. The editor is now full of features that manage paste (and paste-like behaviour) in different scenarios, so returning to the v5 code is not an option.

This issue has turned out to be quite complex. It only happens on Chrome and Safari; the issue is not replicable on Firefox. There doesn't appear to be an easy workaround that works for all scenarios so we might try logging a bug with Google and Apple.

**Comment by TopCoder02** [comment]
I tested this scenario on a few different WYSIWYG Editors, they all functioned properly with this cut and paste scenario in Firefox, Chrome and Safari.  Just providing a little more information, some are open source, it just might be worth taking a look to see what is what the difference is. 

I also tried to find a was to fix it with the on paste event,  I can fix some scenarios' but not all, at least not yet. This is pretty critical functionality to be able to cut and paste accurately.

**Comment by TheSpyder** [comment]
I don't go around browsing other editor codebases. That's a legal grey area for someone in my position. And we already know the browser default `contenteditable` behaviour doesn't replicate this issue; it removes the `font-size: 0` from the outer div and the text appears normally. Perhaps other editors are just not attempting to retain as much of the format tree as TinyMCE does.

We have a replication case of how we treat inserted data without TinyMCE involved (paste is just the way you found it). The case doesn't break on Firefox. Once we've logged it with Chromium and Webkit I'll post the links here.

> This is pretty critical functionality to be able to cut and paste accurately.

Our paste routine does behave accurately in normal scenarios. You've just found an edge case.

**Comment by TopCoder02** [comment]
@TheSpyder 

I just wanted to share one more video,  slightly different scenario.  There's quite a few scenarios, but it's all tied back to the same issue. The video doesn't have voice, I'm getting pretty tired, but you'll understand it. Just watch how the colors change based on where I paste. 

https://fiddle.tiny.cloud/iha7mYU6gi/0

https://github.com/user-attachments/assets/f3614b07-3d95-4b90-9c35-0e2329d304b7

**Comment by TheSpyder** [comment]
I understand your frustration, but once again if you try those steps in Firefox the colour remains as expected.

In order to insert content we use a browser API to delete the selected text, and on Chrome/Safari this does not maintain the cursor position correctly. I don't know if this is a recent change or something that has happened for years and not caught our attention. I did some experiments last week and it doesn't seem possible to work around this issue without breaking other features of the editor.

**Comment by TopCoder02** [comment]
@TheSpyder   

  Sounds like Blink and Webkit based browsers it doesn't work on and it only works on Gecko based browsers? Is my understanding correct?

**Comment by TheSpyder** [comment]
Yes. I'll share the details once we've opened issues with the relevant open source projects.

**Comment by TopCoder02** [comment]
@TheSpyder  - It's been 7 Month since you said you'd file bugs with the relevant open source projects and share the links here. Can you provide those bug report links so the community can track the browser issues?

**Comment by TheSpyder** [comment]
I admit this slipped off my to-do list and I didn't notice. I even had one of our devs create a small replication case for me that doesn't involve TinyMCE so it's better for logging the issue.

Sorry for the delay, I'll get it done soon.

**Comment by TheSpyder** [comment]
I've been digging in to our replication case because it now seems to behave the same on Chrome and Firefox, although the tinymce behaviour hasn't changed. That's frustrating, I'll need to build a new one.
